syntax = "proto3";

package memory;

option go_package = "local/memorypb";

service UserMemoryService {
  rpc Set (SetRequest) returns (SetResponse);
  rpc Get (GetRequest) returns (GetResponse);
  rpc GetUserContext (UserContextRequest) returns (UserContextResponse);
}

message SetRequest {
  string user_id = 1;
  string key = 2;
  string value = 3;
}

message SetResponse {
  bool success = 1;
}

message GetRequest {
  string user_id = 1;
  string key = 2;
}

message GetResponse {
  string value = 1;
  bool found = 2;
}

message UserContextRequest {
  string user_id = 1;
}

message UserContextResponse {
  string context = 1;
}
